src/unix/process.c

```diff
@@ -1012,6 +1013,7 @@ int uv_spawn(uv_loop_t* loop,
       process->flags |= UV_HANDLE_REAP;
       loop->flags |= UV_LOOP_REAP_CHILDREN;
     }
+    loop->nfds++;
```

Comment:
This is to placate the `if (loop->nfds == 0)` check at the top of `uv__io_poll()`? I'd check explicitly whether we're also interested in receiving process events rather than implicitly like this.

Reply:
Meaning that we should change that check to also iterate the process handles to count how many to add to nfds? That seems more complicated than doing it once here

Reply:
Checking `UV_LOOP_REAP_CHILDREN` is probably sufficient? `loop->nfds` is really just used as a boolean in `uv__io_poll()`.

Reply:
That would incorrectly move this inside the if condition above

Reply:
Although maybe all process handles are active, so we would only need to check if that list was empty? (I think this was handled before by the signal pipe watcher counting as an nfds that is never stopped once activated?)

Reply:
> That would incorrectly move this inside the if condition above

Fair point. Can you add a comment here and above then?

Reply:
What did you want it to say?

Reply:
A short explanation that it's to prevent `uv__io_poll()` from bailing out prematurely.

Reply:
I also double-checked, and see that `uv_signal_stop` is called in the close cleanup. I pushed some code cleanup along with the comment so we can avoid doing some redundant work.